projects/igniteui-angular/src/lib/dialog/dialog.component.spec.ts

```diff
@@ -360,6 +361,71 @@ describe('Dialog', () => {
         expect(childDialog.isOpen).toEqual(false);
     }));
 
+    it('Should chenaged closeOnOutsideSelect property while the dialog is open', fakeAsync(() => {
+        const fixture = TestBed.createComponent(ChangeClosePropertyOnOpenDialogComponent);
+        fixture.detectChanges();
+
+        const dialog = fixture.componentInstance.dialog;
+        const dialogElem = fixture.debugElement.query(By.css('.igx-dialog')).nativeElement;
+
+        dialog.open();
+        tick();
+        fixture.detectChanges();
+
+        dialogElem.click();
+        tick();
+        fixture.detectChanges();
+        
+        expect(dialog.isOpen).toEqual(false);
+
+        dialog.open();
+        tick();
+        fixture.detectChanges();
+
+        dialog.closeOnOutsideSelect = false;
+        tick();
+        fixture.detectChanges();
+
+        dialogElem.click();
+        tick();
+        fixture.detectChanges();
+
+        expect(dialog.isOpen).toEqual(true);
+    }));
+
+    it('Should chenaged closeOnEscape property while the dialog is open', fakeAsync(() => {
```

Comment:
extend the test, or add another one, to check in opposite direction if `closeOnEscape` is changed from `false` to `true` while the dialog is open.